src/mlpack/core/cereal/main.cc

```diff
@@ -0,0 +1,31 @@
+#include "pointer_wrapper.hpp"
+#include <fstream>
+
+int main(int argc, char* argv[])
+{
+
+  std::ifstream is("data.xml");
+  cereal::XMLInputArchive archive(is);
+  
+  // std::ofstream is("data.xml");
+  // cereal::XMLOutputArchive archive(is);
+  
+  double data = 40;
+  double* z = nullptr;
+
+  //ref_to_pointer(d); 
+
+  cereal::pointer_wrapper<double> d = cereal::make_pointer(z);
+  archive(d);
+  z = d.release();
```

Comment:
So, in the implementation that you have in `pointer_wrapper.hpp`, it looks like `d.release()` will just return the pointer `localPointer` that's held in `d`.  If `cereal::make_pointer` takes a `T*&` instead of a `T*`, then I think that `d.localPointer` will be exactly equal to `z`, and you could just do this:

```
archive(cereal::make_pointer(z));
```

and that would work for both saving and loading.  It would work for loading because the `pointer_wrapper` is holding a reference to the pointer `z`, so when you call `smartPointer.release()`, that should be directly modifying `z`. :+1:

Reply:
@rcurtin. I agree, but I was getting segmentation fault doing when passing the reference pointer. Finally, I resolved that using the `release` function. Maybe I was missing something. I will re-look into it.